src/main/java/com/zufar/onlinestore/payment/endpoint/PaymentEndpoint.java

```diff
@@ -67,14 +59,9 @@ public ResponseEntity<String> getPaymentMethod(@RequestBody @Valid final CreateP
     }
 
     @PostMapping("/event")
-    public ResponseEntity<Void> paymentEventsProcess(
-            @RequestBody @Valid @NotEmpty @NotNull String paymentIntentPayload,
-            @RequestHeader("Stripe-Signature") @Valid @NotEmpty @NotNull String stripeSignatureHeader) throws SignatureVerificationException {
-        if (Objects.isNull(paymentIntentPayload) || Objects.isNull(stripeSignatureHeader)) {
-            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
-        }
+    public ResponseEntity<Void> paymentEventsProcess(@RequestBody @Valid @NotEmpty final String paymentIntentPayload,
```

Comment:
Using the @Valid annotation on a String parameter like paymentIntentPayload doesn't make sense because @Valid is used to trigger the validation checks on the fields of a complex object (e.g., a custom class or a Bean), not a simple object like a String.

Reply:
done